vignettes/baselinenowcast.Rmd

```diff
@@ -202,20 +203,69 @@ pobs <- enw_preprocess_data(
 triangle_full <- pobs$reporting_triangle[[1]]
 head(triangle_full)
 
-# Check that the sum of the rows in the reporting triangle is the
-# same as the sums across report dates by reference date
 triangle_matrix <- triangle_full |>
   select(-`.group`, -reference_date) |>
   as.matrix() |>
   unname()
-triangle_sums <- rowSums(
-  triangle_matrix
-)
-observed_sums <- observed_long |>
-  group_by(reference_date) |>
-  summarise(total_confirmed = max(confirm)) |>
-  mutate(triangle_sums = triangle_sums)
 ```
+To see what this looks like, we can plot it using `ggplot()`.
+```{r make-heatmap-reporting-triangle}
+triangle_df <- as.data.frame(triangle_matrix) |>
+  mutate(time = row_number()) |>
+  pivot_longer(!time,
+    values_to = "count",
+    names_prefix = "V",
+    names_to = "delay"
+  ) |>
+  mutate(delay = as.numeric(delay))
+
+ggplot(
+  triangle_df,
+  aes(x = delay, y = time, fill = count)
+) +
+  geom_tile() +
+  scale_fill_gradient(low = "white", high = "blue") +
+  labs(title = "Reporting triangle", x = "Delay", y = "Time") +
+  theme_bw() +
+  scale_y_reverse()
+```
+
+It's clear from this figure that the bottom left of the reporting triangle
+contains 0s as a result of the pre-processing we have done.
+In this case, we want to assume that we are making the nowcast as of the last
```

Comment:
The words used to desribe the why for this function hint that it could have a more informative name than the mechanics of what it does

Reply:
Are you referring to `replace_lower_right_with_NA()`? My pushback is `observe_triangle()` doesn't tell me at all what its doing, whereas this is very explicit exactly what it will do... and is more general than just reporting triangles?

Reply:
But fair that this might not be what a baseline nowcast user is thinking about, they want to `transform_to_observed_as_of_last_timepoint()` or something very verbose

Reply:
fair enough. I am always wary of functions that you need to explain to lay people in the getting started( as is happening here.)

Reply:
Could argue the last sentence "To make this change, we can use the package function
`replace_lower_right_with_NA()`." isn't needed?